File: metalanguage/src/main/java/graphs/MLGraph.java
package graphs;

import lombok.Getter;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Getter
public class MLGraph {
    private Set<UUID> entities;
    private Set<UUID> ports;
    private Set<UUID> relations;
    @Setter
    private UUID parentId;

    MLGraph(){
        entities = new HashSet<UUID>();
        ports = new HashSet<UUID>();
        relations = new HashSet<UUID>();
    }
}

File: metalanguage-lib/src/main/kotlin/ru/psu/graphs/MLGraph.kt
package ru.psu.graphs

import java.io.Serializable
import java.util.*
import kotlin.collections.HashSet

//Класс графа в MetaLanguage 1.1
class MLGraph(
        val parentId:UUID?, //Идентификатор конструкции-родителя
        val id: UUID, //Идентификатор
        val prototypeId:UUID? //Прототип (если null, то это конструкция в метамодели)
) : Serializable {
    val entities:MutableSet<UUID> = HashSet() //Id сущностей в графе
    val ports:MutableSet<UUID> = HashSet() //Id портов в графе
    val relations:MutableSet<UUID> = HashSet() //Id отношений в графе
}